src/include/sof/common.h

```diff
@@ -114,6 +113,9 @@
 #define container_of(ptr, type, member) \
 	({const typeof(((type *)0)->member)*__memberptr = (ptr); \
 	(type *)((char *)__memberptr - offsetof(type, member)); })
+#define attr_container_of(ptr, type, member, attr) \
+	({const typeof(((type *)0)->member) attr *__memberptr = (ptr); \
+	(type *)((char attr *)__memberptr - offsetof(type, member)); })
```

Comment:
@lyakh can you please add a comment briefly alluding to an example of a compilation issue that justifies this new macro? It's very hard (for me?) to imagine how an attribute can affect pointer arithmetic and I don't think we should have to `git blame` this macro to understand why it's needed. There's certainly no harm having MORE information in the commit message but there should be at least one line alluding to why this is needed at all because it's quite puzzling otherwise.

Reply:
@marc-hb do I understand it correctly that these your two comments are mutually-exclusive? I.e. I have to select which one we need? We need two versions. So I would add the `attr_container_of()` version to SOF for now. Then when the base sparse support https://github.com/zephyrproject-rtos/zephyr/pull/43776 is merged into Zephyr we can add `CONTAINER_OF_ATTR()` there and move SOF over to using Zephyr macros.

Reply:
They were not mutually exclusive because there was zero suggested change in my Zephyr/CONTAINER_OF comment: it was a pure question trying to understand Liam's question. I think I understand now why you perceived them as mutually exclusive.

Either way and wherever and whenever, my recommendation is to add a one-line comment alluding to what compilation issue requires a new macro because it seems far from obvious.

Reply:
@marc-hb done